WalletWasabi/Affiliation/AffiliateServerStatusUpdater.cs

```diff
@@ -0,0 +1,84 @@
+using System.Linq;
+using System.Collections.Immutable;
+using System.Collections.Generic;
+using WalletWasabi.Affiliation.Models;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+
+namespace WalletWasabi.Affiliation;
+
+public class AffiliateServerStatusUpdater : PeriodicRunner
+{
+	private static readonly TimeSpan AffiliateServerTimeout = TimeSpan.FromSeconds(20);
+	private static readonly TimeSpan Interval = TimeSpan.FromMinutes(1);
+
+	public AffiliateServerStatusUpdater(IDictionary<AffiliationFlag, AffiliateServerHttpApiClient> clients)
+		  : base(Interval)
+	{
+		Clients = clients;
+		RunningAffiliateServers = ImmutableList<AffiliationFlag>.Empty;
+	}
+
+	private IDictionary<AffiliationFlag, AffiliateServerHttpApiClient> Clients { get; }
+	private ImmutableList<AffiliationFlag> RunningAffiliateServers { get; set; }
+
+	public ImmutableArray<AffiliationFlag> GetRunningAffiliateServers()
+	{
+		return RunningAffiliateServers.ToImmutableArray();
+	}
+
+	protected override async Task ActionAsync(CancellationToken cancellationToken)
+	{
+		await UpdateRunningAffiliateServersAsync(cancellationToken).ConfigureAwait(false);
+	}
+
+	private static async Task<bool> IsAffiliateServerRunningAsync(AffiliateServerHttpApiClient client, CancellationToken cancellationToken)
+	{
+		try
+		{
+			StatusResponse result = await client.GetStatusAsync(new StatusRequest(), cancellationToken).ConfigureAwait(false);
```

Comment:
@kiminuo note that this is simply an Unit type that is pretty convenient. I introduced it years ago but other devs didn't like it because they were not familiar with it (UI team uses Unit everywhere). So, we could reintroduce the Unit and replace this and other "empty" classes.